distributed/shuffle/_rechunk.py

```diff
@@ -121,38 +119,65 @@ class ShardID(NamedTuple):
     """
 
     #: Index of the new output chunk to which the shard belongs
-    chunk_index: NIndex
-    #: Index of the shard within the multi-dimensional array of shards that will be
+    chunk_index: NDIndex
+    #: Index of the shard within the n-dimensional array of shards that will be
     # concatenated into the new chunk
-    shard_index: NIndex
+    shard_index: NDIndex
 
 
-def rechunk_slicing(
-    old: ChunkedAxes, new: ChunkedAxes
-) -> dict[NIndex, list[tuple[ShardID, NSlice]]]:
-    """Calculate how to slice the old chunks to create the new chunks
+class Split(NamedTuple):
+    """Slice of a chunk that is concatenated with other splits to create a new chunk
 
-    Returns
-    -------
-        Mapping of each old chunk to a list of tuples defining where each slice
-        of the old chunk belongs. Each tuple consists of the index
-        of the new chunk, the index of the slice within the composition of slices
-        creating the new chunk, and the slice to be applied to the old chunk.
+    Splits define how to slice an input chunk on a single axis into small pieces
+    that can be concatenated together with splits from other input chunks to create
+    output chunks of a rechunk operation.
     """
-    from dask.array.rechunk import intersect_chunks
 
-    ndim = len(old)
-    intersections = intersect_chunks(old, new)
-    new_indices = product(*(range(len(c)) for c in new))
+    #: Index of the new output chunk to which this split belongs.
+    chunk_index: int
+
+    #: Index of the split within the list of splits that are concatenated
+    #: to create the new chunk.
+    split_index: int
+
+    #: Slice of the input chunk.
+    slice: slice
+
 
-    slicing = defaultdict(list)
+SplitChunk: TypeAlias = list[Split]
+SplitAxis: TypeAlias = list[SplitChunk]
+SplitAxes: TypeAlias = list[SplitAxis]
 
-    for new_index, new_chunk in zip(new_indices, intersections):
-        sub_shape = [len({slice[dim][0] for slice in new_chunk}) for dim in range(ndim)]
 
-        shard_indices = product(*(range(dim) for dim in sub_shape))
+def split_axes(old: ChunkedAxes, new: ChunkedAxes) -> SplitAxes:
+    """Calculate how to split the old chunks on each axis to create the new chunks
 
-        for shard_index, sliced_chunk in zip(shard_indices, new_chunk):
-            old_index, nslice = zip(*sliced_chunk)
-            slicing[old_index].append((ShardID(new_index, shard_index), nslice))
-    return slicing
+    Parameters
+    ----------
+    old : ChunkedAxes
+        Chunks along each axis of the old array
+    new : ChunkedAxes
+        Chunks along each axis of the new array
+
+    Returns
+    -------
+    SplitAxes
+        Splits along each axis that determine how to slice the input chunks to create
```

Comment:
I think `Splits` here is a technical term (meaning the `Split` object)?

Reply:
"list of :class:`Split`s ..." ? Which should linkify things?

Reply:
We should explicitly mention the sparsity of the output here.